src/components/note/__internal__/note.component.js

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Editor, EditorState } from 'draft-js';
+import tagComponent from '../../../utils/helpers/tags/tags';
+import {
+  StyledNote, NoteContent, InlineControl, Title, Footer, FooterContent
+} from './note.style.js';
+import StatusWithTooltip from './status-with-tooltip';
+
+const Note = (props) => {
+  const {
+    noteContent,
+    width,
+    inlineControl,
+    title,
+    name,
+    createdDate,
+    status,
+    ...rest
+  } = props;
+
+  const [showTooltip, setShowTooltip] = useState(false);
+
+  const renderStatus = () => {
+    if (!status || !status.text || !status.text.length) {
+      return null;
+    }
+
+    const { text, timeStamp } = status;
+
+    return (
+      <FooterContent>
+        <StatusWithTooltip { ...tooltipProps(timeStamp, showTooltip, setShowTooltip) }>
+          { text }
+        </StatusWithTooltip>
+      </FooterContent>
+    );
+  };
+
+  return (
+    <StyledNote
+      width={ width }
+      { ...rest }
+      { ...tagComponent('note', props) }
+    >
+      { title && (
+        <Title>
+          { title }
+        </Title>
+      )}
+
+      { inlineControl && (
+        <InlineControl>
+          { inlineControl }
+        </InlineControl>
+      )}
+
+      <NoteContent>
+        <Editor readOnly editorState={ noteContent } />
+      </NoteContent>
+
+      {(name && createdDate) && (
+        <NoteContent>
+          <Footer>
+            <FooterContent>{ name }</FooterContent>
+            <FooterContent>{ createdDate }</FooterContent>
+            { renderStatus() }
+          </Footer>
+        </NoteContent>
+      )}
+    </StyledNote>
+  );
+};
+
+function tooltipProps(id, showTooltip, setShowTooltip) {
+  return {
+    tooltipMessage: id,
+    tooltipPosition: 'top',
+    tooltipAlign: 'center',
+    tooltipVisible: showTooltip,
+    onMouseOver: () => setShowTooltip(true),
+    onMouseLeave: () => setShowTooltip(false)
+  };
+}
+
+Note.defaultProps = {
+  noteContent: EditorState.createEmpty()
+};
```

Comment:
is it possible to replace 
```js 
Note.defaultProps = {} 
``` 
with
```js 
noteContent = EditorState.createEmpty()
```
at top of the component?

Reply:
Good point, not sure why I didn't do that in the first place 🤦

Reply:
We should be doing this as `defaultProps` is getting deprecated for functional components